paper-input-container.html

```diff
@@ -0,0 +1,275 @@
+    --paper-input-container-input-color: var(--primary-text-color);
+
+  }
+
+</style>
+
+<dom-module id="paper-input-container">
+
+  <style>
+
+    .floated-label-placeholder {
+      mixin(--paper-font-caption);
+    }
+
+    .focused-line {
+      height: 2px;
+
+      -webkit-transform-origin: center center;
+      transform-origin: center center;
+      -webkit-transform: scale3d(0,1,1);
+      transform: scale3d(0,1,1);
+
+      background: var(--paper-input-container-focus-color);
+    }
+
+    .is-highlighted .focused-line {
+      -webkit-transform: none;
+      transform: none;
+      -webkit-transition: -webkit-transform 0.25s;
+      transition: transform 0.25s;
+
+      mixin(--paper-transition-easing);
+    }
+
+    .is-invalid .focused-line {
+      background: var(--paper-input-container-invalid-color);
+
+      -webkit-transform: none;
+      transform: none;
+      -webkit-transition: -webkit-transform 0.25s;
+      transition: transform 0.25s;
+
+      mixin(--paper-transition-easing);
+    }
+
+    .unfocused-line {
+      height: 1px;
+      background: var(--paper-input-container-color);
+    }
+
+    .input-content ::content label,
+    .input-content ::content .paper-input-label {
+      color: var(--paper-input-container-color);
+
+      mixin(--fit);
+      mixin(--paper-font-subhead);
+
+      mixin(--paper-input-container-label);
+    }
+
+    .input-content.label-is-floating ::content label,
+    .input-content.label-is-floating ::content .paper-input-label {
+      -webkit-transform: translate3d(0, -75%, 0) scale(0.75);
+      transform: translate3d(0, -75%, 0) scale(0.75);
+      -webkit-transform-origin: left top;
+      transform-origin: left top;
+      -webkit-transition: -webkit-transform 0.25s;
+      transition: transform 0.25s;
+
+      mixin(--paper-transition-easing);
+    }
+
+    .input-content.label-is-highlighted ::content label,
+    .input-content.label-is-highlighted ::content .paper-input-label {
+      color: var(--paper-input-container-focus-color);
+    }
+
+    .input-content.is-invalid ::content label,
+    .input-content.is-invalid ::content .paper-input-label {
+      color: var(--paper-input-container-invalid-color);
+    }
+
+    .input-content.label-is-hidden ::content label,
+    .input-content.label-is-hidden ::content .paper-input-label {
+      visibility: hidden;
+    }
+
+    .input-content ::content input,
+    .input-content ::content .paper-input-input {
+      position: relative; /* to make a stacking context */
+      outline: none;
+      color: var(--paper-input-container-input-color);
+
+      mixin(--paper-font-subhead);
+    }
+
+    .input-content ::content input {
+      font: inherit;
+      padding: 0;
+      width: 100%;
+      background: transparent;
+      border: none;
+
+      mixin(--paper-input-container-input);
+    }
+
+  </style>
+
+  <template>
+
+    <!-- <template is="x-if" if="[[!noLabelFloat]]"> -->
```

Comment:
Actually it's needed, but `x-if` wasn't working for some reason. Will add a note.